core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -83,6 +89,11 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
   def main(args: Array[String]) {
 
     info("Starting mirror maker")
+    println("WARNING: The default partition assignment strategy of mirror maker will change from 'range' to 'roundrobin' " +
```

Comment:
@hachikuji Thanks a lot for the explanation. What you suggested makes sense (I had missed the fact that multiple assignors can be provided in the consumer config). If I'm not mistaken the overhead you mentioned is [here](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java#L137). I think your suggestion is a proper way to go about the rolling upgrade (assuming changing the default assignor is imminent).